frontend/src/components/NewArrivals.svelte

```diff
@@ -0,0 +1,149 @@
+<script>
+	import {getContext} from 'svelte';
+	import {Loading} from '.';
+	import {getNewArrivals, getStores, LOW_STOCK} from '../utils/store';
+
+	let ctx = getContext('userInfo');
+	let userInfo;
+	(async () => {
+		userInfo = (await ctx) || null;
+		if (!userInfo) return;
+	})();
+
+	let newArrivals = undefined;
+	let stores = [];
+	(async () => {
+		stores = await getStores();
+		newArrivals = await getNewArrivals().catch(e => e);
+		if (Array.isArray(newArrivals))
+			newArrivals.sort((a, b) => a.name.localeCompare(b.name));
+		setTimeout(checkShouldScroll, 0);
```

Comment:
What's the point of using a timeout here?

Reply:
I need to wait for the elements to refresh so the timeout makes it not run until the next cycle at which point it should be loaded.